ts/screens/authentication/cie/CieCardReaderScreen.tsx

```diff
@@ -0,0 +1,250 @@
+  "reading" = "reading",
+  "error" = "error",
+  "completed" = "completed",
+  "waiting_card" = "waiting_card"
+}
+
+type State = {
+  // Get the current status of the card reading
+  readingState: ReadingState;
+  errorMessage: string;
+};
+
+/**
+ *  This screen shown while reading the card
+ */
+class CieCardReaderScreen extends React.PureComponent<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      /* 
+      These are the states that can occur when reading the cie (from SDK)
+      - waiting_card (we are ready for read ->radar effect)
+      - reading (we are reading the card -> progress animation)
+      - error (the reading is interrupted -> progress animation stops and the progress circle becomes red)
+      - completed (the reading has been completed)
+      */
+      readingState: ReadingState.waiting_card,
+      errorMessage: ""
+    };
+  }
+
+  get ciePin(): string {
+    return this.props.navigation.getParam("ciePin");
+  }
+
+  get cieAuthorizationUri(): string {
+    return this.props.navigation.getParam("authorizationUri");
+  }
+
+  private handleCieEvent = async (event: CEvent) => {
+    // TODO: WHAT is it returned if CIE is EXPIRED? we should redirect to CieExpiredOrInvalidScreen
+    // -- it should be CERTIFICATE_EXPIRED
+    switch (event.event) {
+      case "ON_TAG_DISCOVERED":
+        if (this.state.readingState !== ReadingState.reading) {
+          this.setState({ readingState: ReadingState.reading }, () => {
+            Vibration.vibrate(100);
+          });
+        }
+        break;
+      case "ON_TAG_LOST":
+        this.setState(
+          {
+            readingState: ReadingState.error,
+            errorMessage: I18n.t("authentication.cie.card.error.onTagLost")
+          },
+          () => {
+            Vibration.vibrate(100);
+          }
+        );
+        break;
+
+      case "ON_CARD_PIN_LOCKED":
+        await this.stopCieManager();
+        this.props.navigation.navigate(ROUTES.CIE_PIN_TEMP_LOCKED_SCREEN);
+        break;
+
+      case "AUTHENTICATION_ERROR":
+      case "CERTIFICATE_EXPIRED":
+        this.setState(
+          {
+            readingState: ReadingState.error,
+            errorMessage: I18n.t("authentication.cie.card.error.generic")
+          },
+          () => {
+            Vibration.vibrate(100);
+          }
+        );
+        break;
+
+      case "ON_NO_INTERNET_CONNECTION":
+        this.setState(
+          {
+            readingState: ReadingState.error,
+            errorMessage: I18n.t("authentication.cie.card.error.tryAgain")
+          },
+          () => {
+            Vibration.vibrate(100);
+          }
+        );
+        break;
+      case "ON_PIN_ERROR":
+        await this.handleWrongPin(event.attemptsLeft);
+        break;
+      default:
+        break;
+    }
+  };
+
+  private handleCieError = (error: Error) => {
+    this.setState({
+      readingState: ReadingState.error,
+      errorMessage: error.message
```

Comment:
@valentinacala 
We should avoid security details disclosure.
When the PR is ready it could be replaced with a "generic error"